### PR TITLE
Wait for deletion of agent DaemonSet before deleting other resources.

### PR DESCRIPTION
**Pull request by marseel**
Wait for the deletion of agent DaemonSet before deleting other resources.
This allows the agent to clean up cilium annotations added to the node.

Related PR: https://github.com/cilium/cilium/pull/23772

### CODE REVIEW

File: install/uninstall.go [hunk 1]
```diff
@@ -94,25 +99,25 @@ func (k *K8sUninstaller) Uninstall(ctx context.Context) error {
...
 
-	retry:
+func (k *K8sUninstaller) waitForPodsToBeDeleted(ctx context.Context) error {
+	for {
 		pods, err := k.client.ListPods(ctx, k.params.Namespace, metav1.ListOptions{LabelSelector: defaults.AgentPodSelector})
 		if err != nil {
 			return err
 		}
 
 		if len(pods.Items) > 0 {
```
Comment (squeed) [inline install/uninstall.go:117]: unrelated nit: if the context is cancelled while we're sleeping, we won't notice it. Can we fix that?

**Review by squeed — APPROVED** [review]
Looks good, but there's one more cleanup we might as well do while we're here.

**Comment by marseel** [comment]
PTAL, I've also added some logging when error happens or when we successfully delete daemonset.

**Review by ldelossa — APPROVED** [review]
nice add :100:

**Pull request merged**